lib/manageiq/messaging/kafka/topic.rb

```diff
@@ -8,8 +8,9 @@ module Topic
 
         private
 
-        def publish_topic_impl(options)
-          raw_publish(true, *topic_for_publish(options))
+        def publish_topic_impl(messages)
+          handles = messages.collect { |message| raw_publish(*topic_for_publish(message)) }
+          handles.each(&:wait)
```

Comment:
I like the sending and then the waiting - think this is done very nicely.

Also, nicely mimics the `queue` implementation.